modules/configuration/src/main/java/org/apache/ignite/internal/configuration/storage/Data.java

```diff
@@ -44,10 +45,10 @@ public Data(Map<String, ? extends Serializable> values, long changeId) {
     /**
      * Get values.
      *
-     * @return Values.
+     * @return Unmodifiable map with data values.
      */
     public Map<String, ? extends Serializable> values() {
-        return values;
+        return Collections.unmodifiableMap(values);
```

Comment:
Field itself should already be unmodifiable, please do that in constructor instead